### PR TITLE
NAS-124387 / None / hide private=True fields from query endpoints for non-admins

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

File: src/middlewared/middlewared/plugins/mail.py [hunk 1]
```diff
@@ -93,7 +93,7 @@ class Config:
         Str('security', enum=['PLAIN', 'SSL', 'TLS'], required=True),
         Bool('smtp', required=True),
         Str('user', null=True, required=True),
-        Password('pass', null=True, required=True),
+        Password('pass', null=True, private=True, required=True),
```
Comment (yocalebo) [inline src/middlewared/middlewared/plugins/mail.py:96]: `Password` class already has `private=True` ?

**Review by yocalebo — CHANGES_REQUESTED** [review]
Looks like conflicts need to be resolved.

**Comment by themylogin** [comment]
time 40:00

**Comment by themylogin** [comment]
https://ci.tn.ixsystems.net/jenkins/job/TrueNAS%20SCALE%20-%20Unstable/job/API%20Tests%20-%20TrueNAS%20SCALE%20with%20middleware%20reinstalled%20from%20a%20branch%20Incremental/1507/

**Comment by yocalebo** [comment]
> https://ci.tn.ixsystems.net/jenkins/job/TrueNAS%20SCALE%20-%20Unstable/job/API%20Tests%20-%20TrueNAS%20SCALE%20with%20middleware%20reinstalled%20from%20a%20branch%20Incremental/1507/

133 failed tests so hard to approve this for now.

**Comment by yocalebo** [comment]
Closer inspection seems like the failures are related to this PR: https://github.com/truenas/middleware/pull/12673

**Comment by themylogin** [comment]
https://ci.tn.ixsystems.net/jenkins/job/TrueNAS%20SCALE%20-%20Unstable/job/API%20Tests%20-%20TrueNAS%20SCALE%20with%20middleware%20reinstalled%20from%20a%20branch%20Incremental/1521/

**Review by yocalebo — APPROVED** [review]

**Pull request merged**